adding openai documentation (#938)
microsoft/FLAML: website/docs/Examples/Integrate - OpenAI.md

> FLAML has integrated the OpenAI's completion API.

Reviewer: Readers without background of FLAML could be reading this if they visit this page from reading the paper. Please think of what to say here to keep them interested.

### Prerequisites

> Install the [openai] option.

Reviewer: Please specify the required version number for flaml (1.1.3). And also mention that this feature is subject to change in future.

* num_sumples is the number of different hyperparameter configurations which is allowed to try. The tuning will stop after either num_samples trials or after optimization_budget dollars spent, whichever happens first. -1 means no hard restriction in the number of trials and the actual number is decided by optimization_budget.

Users can specify tuning data, optimization metric, optimization mode, evaluation function, search spaces etc.

```python
config, analysis = oai.Completion.tune(
    data=tune_data,  # the data for tuning
    metric="expected_success",  # the metric to optimize
    mode="max",  # the optimization mode
    eval_func=success_metrics,  # the evaluation function to return the success metrics
    # log_file_name="logs/humaneval.log",  # the log file name
    inference_budget=0.1,  # the inference budget (dollar)
    optimization_budget=4,  # the optimization budget (dollar)
    # num_samples can further limit the number of trials for different hyperparameter configurations;
    # -1 means decided by the optimization budget only
    num_samples=-1,
    model=tune.choice(
        [

>             # These two models are currently free to use from OpenAI,
>             # so no actual cost will incur. They are not free in Azure OpenAI.

Reviewer: ```suggestion
            # These two models are in Beta test and free to use from OpenAI as of Feb 2023,
            # so no actual cost will incur (please double check when you run it). They are not free in Azure OpenAI.
```

>             # The optimization is based on the price in Azure OpenAI.

Reviewer: ```suggestion
            # The optimization is based on the price in Azure OpenAI as of Feb 2023.
```

            "code-cushman-001",
            "code-davinci-002",
        ]
    ),
    prompt=[
        "{prompt}",
        "# Python 3{prompt}",
        "Complete the following Python function:{prompt}",
        "Complete the following Python function while including necessary import statements inside the function:{prompt}",
    ],  # the prompt templates to choose from
    stop=["\nclass", "\ndef", "\nif", "\nprint"],  # the stop sequence
)
```

#### Output tuning results

> After the tuning, we can print out the config and the result found by FLAML:

Reviewer 2: ```suggestion
After the tuning, we can print out the optimized config and the result found by FLAML:
```

```python
print("optimized config", config)
print("best result on tuning data", analysis.best_result)
```

#### Make a request with the tuned config

> We can apply the tuned config on the request for an example task:

Reviewer: ```suggestion
We can apply the tuned config to the request for an instance:
```

> FLAML has integrated the OpenAI's completion API. In this example, we will tune several hyperparameters including the temperature, prompt and n to optimize the inference performance of OpenAI's completion API for a code generation task. Our study shows that tuning hyperparameters can significantly affect the utility of the OpenAI API. For more details please refer to our paper: http://arxiv.org/

Reviewer 2: The link to the paper is not valid.

> ### Prerequisites

Reviewer: ```suggestion

### Prerequisites
```

> Install the [openai] option. You should install version 1.1.3, this version is subject to change from time to time.

Reviewer: ```suggestion
Install the [openai] option. The option is available in flaml since version 1.1.3. This feature is subject to change in future versions.
```

```bash
pip install "flaml[openai]==1.1.3"
```

Setup your OpenAI key:
```python
import os

if "OPENAI_API_KEY" not in os.environ:
    os.environ["OPENAI_API_KEY"] = "<your OpenAI API key here>"
```

> 

Reviewer: "For Azure OpenAI, please ..."
The notebook has information about it. Please add that information here too. Be clear that we support both OpenAI API and Azure OpenAI API.

FLAML has integrated the OpenAI's completion API. In this example, we will tune several hyperparameters including the temperature, prompt and n to optimize the inference performance of OpenAI's completion API for a code generation task. Our study shows that tuning hyperparameters can significantly affect the utility of the OpenAI API.

### Prerequisites

Install the [openai] option. The option is available in flaml since version 1.1.3. This feature is subject to change in future versions.
```bash
pip install "flaml[openai]==1.1.3"
```

Setup your OpenAI key:
```python
import os

if "OPENAI_API_KEY" not in os.environ:
    os.environ["OPENAI_API_KEY"] = "<your OpenAI API key here>"
```

If you use Azure OpenAI, uncomment the following:

Setup your OpenAI key:
```python
# openai.api_type = "azure"
# openai.api_base = "https://<your_endpoint>.openai.azure.com/"
# openai.api_version = "2022-12-01"  # change if necessary

> ```

Reviewer: This is not a notebook, so don't say "uncomment the following" etc. Also, the code snippet is not about "setup key".
